specifications/base.protocol.md

```diff
@@ -0,0 +1,106 @@
+
+How a Self-Issued ID token is obtained from a [=Secure Token Service=] is implementation-specific. An STS may support a
+variety of APIs, including OAuth 2.
+
+#### Using the OAuth 2 Client Credential Grant
+
+A Self-Issued ID Token may be obtained by executing a Client Credential Grant as defined in [[[?rfc6749]]]
+against a [=Secure Token Service=] endpoint. How the participant agent obtains the STS endpoint address is
+implementation-specific and beyond the scope of this specification.
+
+If an `token` is required to be included in the Self Issued ID token, the participant agent will need to request
+scopes for the access token. This is dependent on the API used to obtain the Self-Issued ID Token. An STS implementation
+may use endpoint parameters as defined in the [[[?rfc6749]]] to convey metadata necessary for the creation of the
+`token`
+claim. In this case, the Self-Issued ID Token request may contain a `bearer_access_scope` authorization request
+parameter set to a list of space-delimited scopes the `token` claim will be enabled for. If no `bearer_access_scope`
+parameter is present, the `token` claim should not be included.
+
+<aside class="note">
+A non-normative OpenAPI spec of an STS implementing client credentials flow is provided [here](identity-trust-sts-api.yaml)
+</aside>
+
+### Validating Self-Issued ID Tokens
+
+The Verifier MUST validate the Self-Issued ID Token using the following steps:
+
+1. The Verifier MUST assert that the `iss` and `sub` claims have the same DID value.
+2. The Verifier MUST assert that the `aud` claim is set to the Verifier's DID.
+3. The Verifier MUST validate the signature of the Self-Issued ID token by using a key obtained from the resolved `sub`
+   DID Document. DID resolution is performed according to the DID Method specified by the `sub` claim. The resolved DID
+   document is processed to retrieve the value of the `verificationMethod` property. If the `verificationMethod`
+   property contains one entry and the `kid` token header is not specified, its verification material is used. If more
+   than one entry is present, or the `kid` token header is specified, the verification material from the entry
+   corresponding to the `kid` header is used. If no matching entry is found, the token is rejected.
+4. The Verifier MUST assert that the `sub` claim value equals the `id` property in the DID Document.
+5. The Verifier MUST assert the `nbf` claim if present. The Verifier MAY allow for some leeway to account for clock
+   skew.
+5. The Verifier MUST assert that the current time is before the time represented by the `exp` claim. The Verifier MAY
+   allow for some leeway to account for clock skew. The `iat` claim MAY be used to reject a token that was issued much
+   earlier than the current time, thereby limiting the amount of time a `jti` value needs to be stored.
+6. The Verifier MUST assert that the `jti` claim value has not been used previously.
+
+# Schemas, Contexts, and Message Processing
```

Comment:
If you'd want this to be a separate top-level chapter, you'd have to do another `data-include` in the `index.html`. Currently, it looks like this: 
![image](https://github.com/user-attachments/assets/070961b3-cffb-4f52-85da-9a1eed341021)

Reply:
Should we do that in a subsequent PR?